webClient/src/app/editor/code-editor/monaco/monaco.service.ts

```diff
@@ -307,72 +307,81 @@ export class MonacoService {
+         * time as an enhancement.
+         */
+      if (fileContext.temp) {
         /* Issue a presave check to see if the
          * file can be saved as ISO-8859-1,
          * perhaps this should be done in real
          * time as an enhancement.
          */
         let x = this.preSaveCheck(fileContext);
-        
-        /* The file is temporary, which means that
-         * it was never tagged.
-         */
-        if (fileContext.temp) {
-          /* Open up a dialog with the standard,
+        /* Open up a dialog with the standard,
            * "save as" format.
            */
-          let activeDirectory = '';
-          if (fileDirectory) {
-            activeDirectory = fileDirectory;
-          }
-          let saveRef = this.dialog.open(SaveToComponent, {
-            width: '500px',
-            data: { canBeISO: x, 
-              fileName: fileContext.model.fileName,
-              fileDirectory: activeDirectory }
-          });
-          saveRef.afterClosed().subscribe(result => {
-          if (result) {
-            this.editorControl.saveBuffer(fileContext, result).subscribe(() => obs.next());
-          }
-          });
+        let activeDirectory = '';
+        if (fileDirectory) {
+          activeDirectory = fileDirectory;
         }
-        /* The file was never tagged, so we should
-         * ask the user if they would like to tag
-         * it.
-         */
-        else {
-          /* Open up a dialog asking if the user
-           * wants to tag their file. Again,
-           * we are checking if ISO-8859-1 is
-           * an option.
-           */
-          let saveRef = this.dialog.open(TagComponent, {
-            width: '500px',
-            data: { canBeISO: x,
-                    fileName: fileContext.model.fileName }
-          });
-          saveRef.afterClosed().subscribe(result => {
-          if (result) {
-            this.editorControl.saveBuffer(fileContext, result).subscribe(() => obs.next());
-          }
-          });
+        let saveRef = this.dialog.open(SaveToComponent, {
+          width: '500px',
+          data: { canBeISO: x, 
+            fileName: fileContext.model.fileName,
+            fileDirectory: activeDirectory }
+        });
+        saveRef.afterClosed().subscribe(result => {
+        if (result) {
+          this.editorControl.saveBuffer(fileContext, result).subscribe(() => obs.next());
         }
+        });
+      }
+
+      /* If the file is not new, and the encoding 
+       * has already been set inside of USS via
+       * chtag.
+       */
+      else
+      {
+        this.editorControl.getFileMetadata(fileContext.model.path + '/' + fileContext.model.name).subscribe(r => {
+          fileContext.model.encoding = r.ccsid;
+          if (!fileContext.temp && 
+            fileContext.model.encoding != undefined &&
+            fileContext.model.encoding != null && 
+            fileContext.model.encoding != 0
```

Comment:
Pretty sure if (fileContext.model.encoding) checks for not undefined, not null as truthy values but you should look it up and double check me. So I think all that's needed is

if (fileContext.model.encoding && fileContext.model.encoding != 0)

Reply:
Also, you're setting fileContext.model.encoding = r.ccsid so you can do all of these checks with r.ccsid instead because fileContext.model could be a larger object which you consecutively traverse three times. The performance gains are probably neglible at best, but this would be best practice

Reply:
fixed